packages/obj-logger/src/obj-logger.ts

```diff
@@ -122,7 +122,11 @@ export class ObjLogger implements IObjectLogger {
         let paramsCopy;
 
         if (optionalParams.length) {
-            paramsCopy = JSON.parse(JSON.stringify(optionalParams));
+            try {
+                paramsCopy = JSON.parse(JSON.stringify(optionalParams));
+            } catch {
+                paramsCopy = JSON.parse(JSON.stringify(optionalParams, getCircularReplacer()));
```

Comment:
```suggestion
               throw new Error("Error parsing logger params.");
```

Reply:
```getCircularReplacer``` is passed as the second argument to ```JSON.stringify``` only if parsing without it returns an error. This way, if the object does not contain circular references, parsing without ```getCircularReplacer``` will be faster.

Reply:
objects that we log and have circular references are really rare cases. We do not log them at all in transform-hub. However, it may happen that they are logged in from the Instance level. For this reason we do not need to check for circular references on every log.